src/context.rs

```diff
@@ -471,18 +474,31 @@ impl PySessionContext {
         name: Option<&str>,
         py: Python,
     ) -> PyResult<PyDataFrame> {
-        // Instantiate pyarrow Table object & convert to batches
-        let table = data.call_method0("to_batches")?;
+        let (schema, batches) =
+            if let Ok(stream_reader) = ArrowArrayStreamReader::from_pyarrow_bound(&data) {
+                // Works for any object that implements __arrow_c_stream__ in pycapsule.
+
+                let schema = stream_reader.schema().as_ref().to_owned();
```

Comment:
It doesn't seem that you use a `Schema` instead of a `SchemaRef`; why not just use
```suggestion
                let schema = stream_reader.schema().clone();
```

Reply:
`create_dataframe` below this does require a `Schema`. Well, really it needs an `Option<PyArrowType<Schema>>` which needs ownership.